homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -157,20 +188,136 @@ def device_state_attributes(self):
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return true if entity is at home."""
         try:
-            return bool(self._state["where"] == SureLocationID.INSIDE)
+            return bool(SureLocationID(self._state["where"]) == SureLocationID.INSIDE)
         except (KeyError, TypeError):
             return False
+
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, Any]]:
+        """Return the state attributes of the device."""
+        attributes = None
+        if self._state:
+            try:
+                attributes = {
+                    "since": str(
+                        datetime.fromisoformat(self._state["since"]).replace(
+                            tzinfo=None
+                        )
+                    ),
+                    "where": SureLocationID(self._state["where"]).name.capitalize(),
+                }
+
+            except (KeyError, TypeError) as error:
+                _LOGGER.error(
+                    "Error getting device state attributes from %s: %s\n\n%s",
+                    self._name,
+                    error,
+                    self._state,
+                )
+
+        return attributes
+
+    async def async_update(self) -> None:
+        """Get the latest data and update the state."""
+        self._spc_data = self._spc.states[self._sure_type].get(self._id)
+        self._state = self._spc_data.get("position")
+        _LOGGER.debug("%s -> self._state: %s", self._name, self._state)
+
+
+class DeviceConnectivity(SurePetcareBinarySensor):
+    """Sure Petcare Pet."""
+
+    def __init__(
+        self: BinarySensorDevice,
+        _id: int,
+        sure_type: SureProductID,
+        spc: SurePetcareAPI,
+    ) -> None:
+        """Initialize a Sure Petcare Device."""
+        super().__init__(_id, spc, DEVICE_CLASS_CONNECTIVITY, sure_type)
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device if any."""
+        return f"{self._name}_connectivity"
+
+    @property
+    def unique_id(self: BinarySensorDevice) -> str:
+        """Return an unique ID."""
+        return f"spc-{self._spc_data['household_id']}-{self._id}-connectivity"
+
+    @property
+    def available(self) -> bool:
+        return bool(self._state["online"])
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if entity is online."""
+        return self.available
+
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, Any]]:
+        """Return the state attributes of the device."""
+        attributes = None
+        if self._state:
+            try:
+                attributes = {
+                    "device_rssi": f'{self._state["signal"]["device_rssi"]:.2f}',
+                    "hub_rssi": f'{self._state["signal"]["hub_rssi"]:.2f}',
+                }
+
+            except (KeyError, TypeError) as error:
+                _LOGGER.error(
```

Comment:
There should be no side effects in entity state properties.

Reply:
done 👍